addons/account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -192,7 +192,8 @@ def _complete_stmts_vals(self, stmts_vals, journal, account_number):
                     # reconciliation process will be linked to the bank when the statement is closed.
                     identifying_string = line_vals.get('account_number')
                     if identifying_string:
-                        partner_bank = self.env['res.partner.bank'].search([('acc_number', '=', identifying_string)], limit=1)
+                        partner_bank = self.env['res.partner.bank']\
+                            .search([('acc_number', '=', identifying_string), ('company_id', 'in', (False, self.env.company.id))], limit=1)
```

Comment:
For more robustness, I would prefer journal.company_id.id instead of self.env.company.id